controllers/configurationpolicy_controller.go

```diff
@@ -1703,7 +1703,18 @@ func (r *ConfigurationPolicyReconciler) handleSingleObj(
 			var uid string
 			completed, reason, msg, uid, err := r.enforceByCreatingOrDeleting(obj)
 
-			result.events = append(result.events, objectTmplEvalEvent{completed, reason, msg})
+			hasStatus := false
+			if tmplObj, err := unmarshalFromJSON(objectT.ObjectDefinition.Raw); err == nil {
+				_, hasStatus = tmplObj.Object["status"]
+			}
+
+			if completed && hasStatus {
+				msg += ", but has a status that might not be correct yet"
+				reason += ", but has a status that might not be correct yet"
+				result.events = append(result.events, objectTmplEvalEvent{false, reason, msg})
+			} else {
+				result.events = append(result.events, objectTmplEvalEvent{completed, reason, msg})
+			}
```

Comment:
This is the creation flow, right? Does it still stay NonCompliant if the policy is updated and the status doesn't match or does some logic need to be there also?

Reply:
Good catch, somehow I didn't think about the update flow having this same potential issue!

Reply:
It looks like it might work correctly, but I'll adjust the test to be sure.

Reply:
Updated test: https://github.com/open-cluster-management-io/config-policy-controller/compare/eceb957c80fbc032d898a463b02be38fbd6fa600..3421b2310929e5f8b3e6aafc17034c1352487cc3

Reply:
Initially, the CI here failed, but from the logs it looked like that was because the config-policy-controller was a little overwhelmed with the parallel tests. Because this test uses the status of *another* configurationpolicy, it can take extra loops to do what is expected... so bumping the timeouts to the default (1 minute) fixed it.